src/Forms/GridField/GridFieldFilterHeader.php

```diff
@@ -491,4 +505,36 @@ private function getTitle(object $inst): string
 
         return ClassInfo::shortName($inst);
     }
+
+    /**
+     * Transform search context into basic search context (preserve all relevant search settings)
+     *
+     * @param GridField $gridField
+     * @param SearchContext $searchContext
+     * @return BasicSearchContext
+     */
+    private function getBasicSearchContext(GridField $gridField, SearchContext $searchContext): BasicSearchContext
+    {
+        // Retrieve filters settings as these can be carried over as is
+        $defaultSearchFields = $searchContext->getSearchFields();
+        $defaultFilters = $searchContext->getFilters();
+        $list = $gridField->getList();
+
+        // Carry over any search form settings
+        $basicSearchContext = BasicSearchContext::create($list->dataClass());
+        $basicSearchContext->setFields($defaultSearchFields);
+
+        // Carry over filter configuration (make changes to filter classes so they work with ArrayList data)
+        foreach ($defaultFilters as $defaultFilter) {
+            $fieldFilter = PartialMatchFilter::create(
+            // Use name instead of full name as this plain filter doesn't understand relations
+                $defaultFilter->getName(),
```

Comment:
```suggestion
                // Use name instead of full name as this plain filter doesn't understand relations
                $defaultFilter->getName(),
```